analyses/molecular-subtyping-EPN/02_ependymoma_generate_all_data.py

```diff
@@ -0,0 +1,179 @@
+
+count=0
+
+# Reading in gene set enrichment analyses file for GSEA scores for NKKB pathway
+gsva = pd.read_csv("analyses/gene-set-enrichment-analysis/results/gsva_scores_stranded.tsv", sep="\t")
+gsva_NFKB = gsva.loc[gsva['hallmark_name'] == "HALLMARK_TNFA_SIGNALING_VIA_NFKB"]
+gsva_NFKB = gsva_NFKB.set_index("Kids_First_Biospecimen_ID")
+
+# Reading collapsed gene expression file, none in polya file, so ignoring that file 
+readRDS = robjects.r["readRDS"]
+fpkm_df = readRDS("analyses/molecular-subtyping-EPN/epn-subset/epn_pbta-gene-expression-rsem-fpkm-collapsed.stranded.rds")
+
+#Reading fusion summary file
+fusion = pd.read_csv("analyses/fusion-summary/results/fusion_summary_ependymoma_foi.tsv", sep="\t")
+fusion = fusion.set_index("Kids_First_Biospecimen_ID")
+
+#Reading chromosomal instability file for breakpoint density 
+breakpoint_density = pd.read_csv("analyses/chromosomal-instability/breakpoint-data/union_of_breaks_densities.tsv", sep="\t")
+breakpoint_density = breakpoint_density.set_index("samples")
+
+# creating empty lists to get mean and standard deviation of gene FPKM's from pbta-gene-expression-rsem-fpkm-collapsed.stranded.rds file
+#BSID_list = []
+RELA_fpkm = []
+L1CAM_fpkm = []
+ARL4D_fpkm = []
+CLDN1_fpkm = []
+CXorf67_fpkm = []
+TKTL1_fpkm = []
+GPBP1_fpkm = []
+IFT46_fpkm = []
+
+
+# Opening the FPKM file separately to fill up the mean and stdev lists 
+with open("analyses/molecular-subtyping-EPN/results/EPN_molecular_subtype.tsv", "r") as notebook:
+    for line in notebook:
+        line = line.split()
+        if not line[0].startswith("Kids"):
+            RELA_fpkm.append(fpkm_df.get_value("RELA", line[3]))
+            L1CAM_fpkm.append(fpkm_df.get_value("L1CAM", line[3]))
+            ARL4D_fpkm.append(fpkm_df.get_value("ARL4D", line[3]))
+            CLDN1_fpkm.append(fpkm_df.get_value("CLDN1", line[3]))
+            CXorf67_fpkm.append(fpkm_df.get_value("CXorf67", line[3]))
+            TKTL1_fpkm.append(fpkm_df.get_value("TKTL1", line[3]))
+            GPBP1_fpkm.append(fpkm_df.get_value("GPBP1", line[3]))
+            IFT46_fpkm.append(fpkm_df.get_value("IFT46", line[3]))                  
+                                
+
+# Generating mean and stdev scores from the FPKM list
+RELA_mean = np.mean(RELA_fpkm)
+RELA_stdev = np.std(RELA_fpkm)
+L1CAM_mean = np.mean(L1CAM_fpkm)
+L1CAM_stdev = np.std(L1CAM_fpkm)
+ARL4D_mean = np.mean(ARL4D_fpkm)
+ARL4D_stdev = np.std(ARL4D_fpkm)
+CLDN1_mean = np.mean(CLDN1_fpkm)
+CLDN1_stdev = np.std(CLDN1_fpkm)
+CXorf67_mean = np.mean(CXorf67_fpkm)
+CXorf67_stdev = np.std(CXorf67_fpkm)
+TKTL1_mean = np.mean(TKTL1_fpkm)
+TKTL1_stdev = np.std(TKTL1_fpkm)
+GPBP1_mean = np.mean(GPBP1_fpkm)
+GPBP1_stdev = np.std(GPBP1_fpkm)
+IFT46_mean = np.mean(IFT46_fpkm)
+IFT46_stdev = np.std(IFT46_fpkm)
+
+
+# Looping through EPN sample file with both RNA and DNA BSID's   
+with open("analyses/molecular-subtyping-EPN/results/EPN_molecular_subtype.tsv", "r") as notebook:
```

Comment:
here too, I think you may be more efficient to read the whole file in with pandas, rather than going through it line by line; that can also save you a lot of casting strings to ints that you are doing manually now.